pkg/platform/provider/baremetal/phases/kubeadm/kubeadm.go

```diff
@@ -298,19 +300,31 @@ func UpgradeNode(s ssh.Interface, client kubernetes.Interface, platformClient pl
 	if !needUpgrade {
 		return upgraded, nil
 	}
-
-	// Step 1: drain node
-	err = drainNodeCarefully(s, client, node.Name, option.MaxUnready)
+	// check node kubelet version
+	sameMinor, err := checkKubeletVersion(client, node.Name, option.Version, false)
```

Comment:
pass the `node`  obj to fun checkKubeletVersion will no need api call to get it again